core/schema/util.go

```diff
@@ -73,3 +74,47 @@ var AllVersion = core.AllVersion
 
 type BeforeVersion = core.BeforeVersion
 type AfterVersion = core.AfterVersion
+
+// Return a list of paths to include .gitignore files rebased on parentPath from hostPath
+//
+// Example:
+//
+//   - parentPath = "/foo/bar"
+//   - hostPath = "/foo/bar/baz"
+//   - .gitignore files will be loaded from the following paths: [.gitignore, baz/**.gitignore]
+//
+// ---
+//   - parentPath = "/"
+//   - hostPath = "/foo/bar"
+//   - .gitignore files will be loaded from the following paths: [.gitignore, foo/.gitignore, foo/bar/**.gitignore]
+//
+// We assume the hostPath is always a child of the parentPath.
+func getGitIgnoreIncludePaths(parentPath string, hostPath string) ([]string, error) {
+	var paths []string
+
+	current := hostPath
+	for {
+		if current == parentPath {
```

Comment:
This looks brittle if they're formatted slightly differently like if `parentPath` has a slash in the end this wouldn't be hit. I suggest working from the relative path between them instead.

Reply:
Done in [5955331](https://github.com/dagger/dagger/pull/10736/commits/5955331f945363a279125a5531bd42a63900bf10)